frontend/src/pages/create_campaign/index.js

```diff
@@ -1,34 +1,50 @@
-import DateTimePicker from '@mui/lab/DateTimePicker';
-import React from "react";
-import { useNavigate } from "react-router-dom";
-import { CampaignContainer, CampaignDropzone, CampaignRowDiv, CampaignSubmit, CampaignTextField } from "./createCampaign.styled";
-import Dropzone from 'react-dropzone';
-import { LoadingIndicator } from "../../components";
-import { fileToDataUrl } from "../../utils";
+import React, { useState } from "react";
+import { Container, Tabs, Tab } from "@mui/material";
+import CampaignTab from "./Campaign";
+import RolesTab from "./Roles";
+import CampaignReview from "./Preview";
+import { NextButton, ArrowIcon, NextWrapper } from "./createCampaign.styled";
 
+// FIXME: CHAOS-66, user authentication and redirection if they are not logged in or authenticated
 const CreateCampaign = () => {
-  const navigate = useNavigate();
-
-  const [name, setName] = React.useState("");
-  const [startDate, setStartDate] = React.useState(new Date());
-  const [endDate, setEndDate] = React.useState(new Date());
-  const [description, setDescription] = React.useState("");
-  const [interviewStage, setInterviewStage] = React.useState(false);
-  const [scoringState, setScoringStage] = React.useState(false);
-  const [draft, setDraft] = React.useState(false);
-  const [cover, setCover] = React.useState(null);
-  const [error, setError] = React.useState(null);
-
-  // TODO: user authentication and redirection if they are not logged in or authenticated
+  const [tab, setTab] = useState(0);
+  const [campaignName, setCampaignName] = useState("");
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  const [description, setDescription] = useState("");
+  const [interviewStage, setInterviewStage] = useState(false);
+  const [scoringStage, setScoringStage] = useState(false);
+  const [draft, setDraft] = useState(false);
+  const [cover, setCover] = useState(null);
+  const [error, setError] = useState(null);
+  const [roles, setRoles] = useState([]);
+  const [roleSelected, setRoleSelected] = useState(
+    roles.length > 0 ? roles[0].id : "-1"
+  );
+  const [questions, setQuestions] = useState([]);
+  const onTabChange = (val) => {
+    // only allow user to access review/publish tab if all inputs are non-empty
+    if (val === 2) {
+      if (
+        campaignName === "" ||
+        description === "" ||
+        cover === null ||
+        questions.length === 0 ||
+        roles.length === 0
+      ) {
+        alert("All fields must be filled before reviewing!");
+        return;
+      }
+    }
+    setTab(val);
+  };
 
-  const submitHandler = async () => {
-    if (name.length === 0 && !draft) {
+  // FIXME: CHAOS-64, update submitHandler to account for new data
+  //        (roles/questions etc.), part of backend integration
+  const submitHandler = async (isDraft) => {
+    console.log(`submit handler -> isDraft=${isDraft}`);
+    /*
```

Comment:
is this comment necessary? is it for CHAOS64?

Reply:
The commented out code is an old submitHandler written from campaignCreate before roles/questions etc. were added. I left it in there cause it will be handy when doing CHAOS-64, will just have to add some more error handling + change the request a bit.